provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java

```diff
@@ -102,6 +102,14 @@ public SecurityProviderTPMEmulator(String registrationId) throws SecurityProvide
         srkPublic = createPersistentPrimary(tpm, SRK_PERSISTENT_HANDLE, TPM_RH.OWNER, SRK_TEMPLATE, "SRK");
     }
 
+    public static SecurityProviderTPMEmulator createProviderFromKey(String authenticationKey) throws SecurityProviderException
```

Comment:
This would need both the registration Id and the authentication key.

Reply:
I see that we have a constructor that doesn't take in a registration Id. Is that a valid scenario for DPS? @timtay-microsoft ?

Reply:
[1:43 PM] Tim Taylor
IIRC the registration Id is only relevant when communicating with DPS. After a successful registration, the registration Id isn't needed anymore. So it sounds like Nate is on the right track here